stm32: drivers: flash: mcumgr: support for H7

JvanDooren opened the pull request:
Add support for setting flash word alignment size.
Update to tinycbor 0.5.4 for mcumgr (https://github.com/fancom/tinycbor/pull/1).

Signed-off-by: Jeroen van Dooren <jeroen.van.dooren@nobleo.nl>

JvanDooren committed:
stm32: drivers: flash: mcumgr: support for H7

Add support for setting flash word alignment size.
Update to tinycbor 0.5.4 for mcumgr.

Signed-off-by: Jeroen van Dooren <jeroen.van.dooren@nobleo.nl>

heinwessels commented on drivers/flash/flash_stm32_qspi.c:
```diff
@@ -1094,6 +1094,9 @@ static int flash_stm32_qspi_init(const struct device *dev)
 	LOG_DBG("Device %s initialized", DEV_NAME(dev));
 
+	/* wait until quad output is set before doing an immediate read/write call */
+	qspi_wait_until_ready(dev);
```
A little nit-picky, but this should be before `LOG_DBG("Device %s initialized", DEV_NAME(dev));`. Then it won't say it's initialized if the Flash never gets `ready` for some reason.

JvanDooren replied:
Will do

heinwessels commented on west.yml:
```diff
@@ -119,7 +125,9 @@ manifest:
-      revision: 40daca97b478989884bffb5226e9ab73ca54b8c4
+      repo-path: tinycbor
+      remote: optima
+      revision: 69e745dd9d4975a770ca70195f61510b43e125d4
```
This is not the final hash, the pull request on tinycbor should be merged first.

JvanDooren replied:
Will do

JvanDooren committed:
stm32: drivers: flash: mcumgr: support for H7

Add support for setting flash word alignment size.
Update to tinycbor 0.5.4 for mcumgr.

Signed-off-by: Jeroen van Dooren <jeroen.van.dooren@nobleo.nl>
